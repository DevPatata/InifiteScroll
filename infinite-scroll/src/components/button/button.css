.custom-button {
  border-radius: 8px;
  border: 1px solid transparent;
  padding: 0.6em 1.2em;
  font-size: 1em;
  font-weight: 500;
  font-family: inherit;
  cursor: pointer;
  transition: border-color 0.25s;
}

.custom-button:hover {
  border-color: var(--color-grey);
}

.custom-button:focus,
.custom-button:focus-visible {
  outline: 4px auto -webkit-focus-ring-color;
}

.custom-button-primary {
  background-color: var(--color-light-grey);
}

.custom-button-secondary {
  color: #fff;
  background-color: var(--color-grey);
}

.custom-button-secondary:hover {
  color: var(--color-font);
  background-color: var(--color-light-grey);
}