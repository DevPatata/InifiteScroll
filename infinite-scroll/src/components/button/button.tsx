import React from 'react';
import './button.css';

export function ButtonItem({ action, buttonValue = 'button', isPrimary = true }) {
  return (
    <button data-testid='button-test-id' className={`custom-button ${isPrimary ? 'custom-button-primary' : 'custom-button-secondary'}`} onClick={action}> 
      {buttonValue} 
    </button>
  )
}