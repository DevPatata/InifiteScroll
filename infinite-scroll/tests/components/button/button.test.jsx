import React from 'react';
import { render, screen } from '@testing-library/react';
import { ButtonItem } from '../../../src/components/button/button'

describe('Pruebas en component button', () => {

  test('Prueba match snaptshot', () => {
    const { container } = render(<ButtonItem />);
    expect(container).toMatchSnapshot();
  });

  test('Prueba valor button', () => {
    const buttonValue = 'TestButton';

    render(<ButtonItem buttonValue={'TestButton'} />);

    expect( screen.getByText(buttonValue)).toBeTruthy();

    expect( screen.getByTestId('button-test-id').innerHTML).toContain(buttonValue);
  });

  test('Prueba className primary button', () => {
    const isPrimary = true;

    render(<ButtonItem isPrimary={isPrimary} />);

    expect(screen.getByTestId('button-test-id').className).toContain('custom-button-primary');
  });

  test('Prueba className secondary button', () => {
    const isPrimary = false;

    render(<ButtonItem isPrimary={isPrimary} />);
    
    expect(screen.getByTestId('button-test-id').className).toContain('custom-button-secondary');
  });
})
