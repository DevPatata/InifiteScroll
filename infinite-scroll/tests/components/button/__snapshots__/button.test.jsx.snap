// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en component button Prueba match snaptshot 1`] = `
<div>
  <button
    class="custom-button custom-button-primary"
    data-testid="button-test-id"
  >
    button
  </button>
</div>
`;
